data/policies/derive.cas

```diff
@@ -0,0 +1,29 @@
+virtual resource foo {
+	fn read(domain source) {
+		allow(source, this, file, read);
+	}
+}
+
+virtual resource bar {
+	fn read(domain source) {
+		allow(source, this, dir, read);
+	}
+}
+
+resource custom_define inherits foo, bar {
+	fn read(domain source) {
+		allow(source, this, lnk_file, read);
+	}
+}
+
+@derive([read], strategy=union)
```

Comment:
I feel like the "empty list" approach seems like just as much of an exception as adding a keyword for it.  I do hear your argument that it's weird that this is sometimes a type name and sometimes a keyword, but I don't really see using [] as any better.  I think the average person encountering that for the first time would be more likely to assume it meant a noop function rather than unioning all.

What if we used one `all` keyword for both strategy and function, and changed `strategy` to union?  So you could say something like `@derive([all], [all])`.  That would at least be more internally consistent.

> Or maybe interpret no explicit strategy as an union?

Well, that already happens because "union" is the default value of the argument.  But there should still be some way to refer to it explicitly.

Another factor here is that we've discussed a few times the possibility of someday having other strategies for how derivation works.  I don't have any particular ideas for what strategies we might employ, but I'm a little hesitant to define a syntax now that makes adding mores strategies later in a backwards compatible way more difficult.

Reply:
What about `derive_union()`? Do you think other strategy would make sense?

Reply:
I don't see what `derive_union` achieves?  It just seems more verbose to me.